File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tag-stylebackground-color-rgba0-10-32-005-darkreader-inline-bgcolor-rgba0-8-26-005-data-darkreader-inline-bgcolordivgoogle-divdiv-classcompanytagscontainer-tagoccurence3-div-divdiv-classcompanytagscontainer-tag-stylebackground-color-rgba0-10-32-005-darkreader-inline-bgcolor-rgba0-8-26-005-data-darkreader-inline-bgcolordivamazon-divdiv-classcompanytagscontainer-tagoccurence2-div-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-darkreader-inline-fill-4aa0f2-data-darkreader-inline-fillpolygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div1658-minimum-operations-to-reduce-x-to-zero/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tag-stylebackground-color-rgba0-10-32-005-darkreader-inline-bgcolor-rgba0-8-26-005-data-darkreader-inline-bgcolordivgoogle-divdiv-classcompanytagscontainer-tagoccurence3-div-divdiv-classcompanytagscontainer-tag-stylebackground-color-rgba0-10-32-005-darkreader-inline-bgcolor-rgba0-8-26-005-data-darkreader-inline-bgcolordivamazon-divdiv-classcompanytagscontainer-tagoccurence2-div-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-darkreader-inline-fill-4aa0f2-data-darkreader-inline-fillpolygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div1658-minimum-operations-to-reduce-x-to-zero.java
class Solution {
    public int minOperations(int[] nums, int x) {
        int total=0;
        for(int i=0; i<nums.length; i++) total+= nums[i];
        int sumToRemove= total-x;
        int eleToRemove=0, currSum=0, l=0;
        if(sumToRemove == 0) return nums.length;
        for(int i=0; i<nums.length; i++){
            currSum+= nums[i];
            while(currSum> sumToRemove && l<i){
                currSum-= nums[l];
                l++;
            }
            if(currSum==sumToRemove){
                eleToRemove = Math.max(eleToRemove, i-l+1);
            }
        }
        if(eleToRemove==0) return -1;
        return nums.length- eleToRemove;
    }
}

File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tag-stylebackground-color-rgba0-10-32-005-darkreader-inline-bgcolor-rgba0-8-26-005-data-darkreader-inline-bgcolordivgoogle-divdiv-classcompanytagscontainer-tagoccurence3-div-divdiv-classcompanytagscontainer-tag-stylebackground-color-rgba0-10-32-005-darkreader-inline-bgcolor-rgba0-8-26-005-data-darkreader-inline-bgcolordivamazon-divdiv-classcompanytagscontainer-tagoccurence2-div-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-darkreader-inline-fill-4aa0f2-data-darkreader-inline-fillpolygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div1658-minimum-operations-to-reduce-x-to-zero/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tag-stylebackground-color-rgba0-10-32-005-darkreader-inline-bgcolor-rgba0-8-26-005-data-darkreader-inline-bgcolordivgoogle-divdiv-classcompanytagscontainer-tagoccurence3-div-divdiv-classcompanytagscontainer-tag-stylebackground-color-rgba0-10-32-005-darkreader-inline-bgcolor-rgba0-8-26-005-data-darkreader-inline-bgcolordivamazon-divdiv-classcompanytagscontainer-tagoccurence2-div-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-darkreader-inline-fill-4aa0f2-data-darkreader-inline-fillpolygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div1658-minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int total=accumulate(nums.begin(),nums.end(),0),sum=total-x,maxlen=0,currsum=0,i=0;
        if(sum==0) return nums.size();
        for(int j=0; j<nums.size(); j++){
            currsum+= nums[j];
            while(i<nums.size() && currsum>sum) currsum-= nums[i++];
            if(currsum == sum) maxlen= max(maxlen, j-i+1);
        }
        if(!maxlen) return -1;
        return nums.size()-maxlen;
    }
};